### DISCUSSION TITLE
How/Can we filter product by attribute values?

### DISCUSSION THREAD

**Discussion by g-londhe** [saleor/saleor#10197] [Q&A]
Query
`query ProductList($first: Int, $filter: ProductFilterInput) {
  products(first: $first, filter: $filter) {
    totalCount
    edges {
      node {
        id
        name
        attributes{
          attribute{
            id
            name
            slug
          }
          values{
            name
          }
        }
      }
    }
  }
}`

Variables

`{
  "first": 1,
  "filter": {
    "attributes": {
      "slug": "attribute-slug-here",
      "values": "attribute-value-here"
    }
  }
}`

I'm getting following error for the same, would like to know more how can I update the above query and get desired result.

`        "message": "Variable \"$filter\" got invalid value {\"attributes\": {\"slug\": \"tag-number\", \"values\": \"12345\"}}.\nIn field \"attributes\": In element #0: Expected \"AttributeInput\", found not an object.\nIn field \"attributes\": In element #1: Expected \"AttributeInput\", found not an object."`

**Comment by karolkielecki** [comment]
Hi @g-londhe, please change query variables to 

`{ "first": 1, "filter": { "attributes": [{ "slug": "attribute-slug-here", "values": "attribute-value-here" }] } }` 

and let me know how it works. Cheers!

**Reply by g-londhe** [reply]
Hi @karolkielecki,

I have tried it for rtf-type and plan-text type as attribute value, output returning null.

Is there any way, I can check it out online, it it's working fine?

FYI, I'm using following version of saleor,

saleor => 3.5.3

**Reply by g-londhe** [reply]
Also above method you shared is working for numeric type like =>

```{
  "first": 1,
  "filter": {
    "attributes": [{
      "slug": "attribute-slug-value",
      "valuesRange": {
          "gte": "12345",
          "lte": "12345"
      }
    }]
  }
}
```

**Reply by g-londhe** [reply]
I have added one more bug regarding the same here -> #10094

**Reply by karolkielecki** [reply]
@g-londhe It seems there is a bug there -- filtering doesn't work for some attribute types. We've assigned a ticket to our devs to investigate and fix it. If I have any news, I'll keep you posted.
